pkg/alibabacloud/api/api.go

```diff
@@ -0,0 +1,501 @@
+}
+
+// DescribeNetworkInterface get ENI by id
+func (c *Client) DescribeNetworkInterface(ctx context.Context, eniID string) (*ecs.NetworkInterfaceSet, error) {
+	req := ecs.CreateDescribeNetworkInterfacesRequest()
+	req.NetworkInterfaceId = &[]string{eniID}
+	resp, err := c.ecsClient.DescribeNetworkInterfaces(req)
+	if err != nil {
+		return nil, err
+	}
+	if len(resp.NetworkInterfaceSets.NetworkInterfaceSet) == 0 {
+		return nil, fmt.Errorf("failed to find eni %s", eniID)
+	}
+	return &resp.NetworkInterfaceSets.NetworkInterfaceSet[0], nil
+}
+
+// CreateNetworkInterface creates an ENI with the given parameters
+func (c *Client) CreateNetworkInterface(ctx context.Context, toAllocate int, vSwitchID string, groups []string, tags map[string]string) (string, *eniTypes.ENI, error) {
+	req := ecs.CreateCreateNetworkInterfaceRequest()
+	req.SecondaryPrivateIpAddressCount = requests.NewInteger(toAllocate)
+	req.VSwitchId = vSwitchID
+	req.SecurityGroupIds = &groups
+	reqTag := make([]ecs.CreateNetworkInterfaceTag, 0, len(tags))
+	for k, v := range tags {
+		reqTag = append(reqTag, ecs.CreateNetworkInterfaceTag{
+			Key:   k,
+			Value: v,
+		})
+	}
+	req.Tag = &reqTag
+
+	c.limiter.Limit(ctx, "CreateNetworkInterface")
+
+	sinceStart := spanstat.Start()
+	resp, err := c.ecsClient.CreateNetworkInterface(req)
+	c.metricsAPI.ObserveAPICall("CreateNetworkInterface", deriveStatus(err), sinceStart.Seconds())
+	if err != nil {
+		return "", nil, err
+	}
+
+	var privateIPSets []eniTypes.PrivateIPSet
+	for _, p := range resp.PrivateIpSets.PrivateIpSet {
+		privateIPSets = append(privateIPSets, eniTypes.PrivateIPSet{
+			Primary:          p.Primary,
+			PrivateIpAddress: p.PrivateIpAddress,
+		})
+	}
+	eni := &eniTypes.ENI{
+		NetworkInterfaceID: resp.NetworkInterfaceId,
+		MACAddress:         resp.MacAddress,
+		Type:               resp.Type,
+		SecurityGroupIDs:   resp.SecurityGroupIds.SecurityGroupId,
+		VPC: eniTypes.VPC{
+			VPCID: resp.VpcId,
+		},
+		ZoneID: resp.ZoneId,
+		VSwitch: eniTypes.VSwitch{
+			VSwitchID: resp.VSwitchId,
+		},
+		PrimaryIPAddress: resp.PrivateIpAddress,
+		PrivateIPSets:    privateIPSets,
+		Tags:             parseECSTags(resp.Tags.Tag),
+	}
+	return resp.NetworkInterfaceId, eni, nil
+}
+
+// AttachNetworkInterface attaches a previously created ENI to an instance
+func (c *Client) AttachNetworkInterface(ctx context.Context, instanceID, eniID string) error {
+	req := ecs.CreateAttachNetworkInterfaceRequest()
+	req.InstanceId = instanceID
+	req.NetworkInterfaceId = eniID
+	c.limiter.Limit(ctx, "AttachNetworkInterface")
+	sinceStart := spanstat.Start()
+	_, err := c.ecsClient.AttachNetworkInterface(req)
+	c.metricsAPI.ObserveAPICall("AttachNetworkInterface", deriveStatus(err), sinceStart.Seconds())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// WaitEniAttached check eni is attached to ECS and return attached ECS instanceID
+func (c *Client) WaitEniAttached(ctx context.Context, eniID string) (string, error) {
```

Comment:
```suggestion
func (c *Client) WaitENIAttached(ctx context.Context, eniID string) (string, error) {
```

While you're at it, please make sure all abbreviations are consistent.